lua/metals.lua

```diff
@@ -173,30 +176,21 @@ M.logs_toggle = function()
 end
 
 -- Implements the new-scala-file feature.
--- https://scalameta.org/metals/docs/integrations/new-editor.html#create-new-scala-file
+-- https://scalameta.org/metals/docs/integrations/new-editor/#create-new-scala-file
 --
 -- @param directory_uri_opt Path URI for the new file. e.g. 'file:///home/...'
 -- @param name_opt Name for the scala file. e.g.: 'MyNewClass'. If nil, it's asked in an input box.
 -- @param file_type_opt Type of file. e.g.: 'worksheet'
 M.new_scala_file = function(directory_uri_opt, name_opt, file_type_opt)
-  local args_string_array = {}
-  if directory_uri_opt then
-    table.insert(args_string_array, 1, directory_uri_opt)
-  else
-    table.insert(args_string_array, 1, vim.NIL)
-  end
+  directory_uri_opt = directory_uri_opt or vim.NIL
+  name_opt = name_opt or vim.NIL
+  file_type_opt = file_type_opt or vim.NIL
 
-  if name_opt then
-    table.insert(args_string_array, 2, name_opt)
-  else
-    table.insert(args_string_array, 2, vim.NIL)
-  end
+  local args_string_array = {}
 
-  if file_type_opt then
-    table.insert(args_string_array, 3, file_type_opt)
-  else
-    table.insert(args_string_array, 3, vim.NIL)
-  end
+  table.insert(args_string_array, 1, directory_uri_opt)
+  table.insert(args_string_array, 2, name_opt)
+  table.insert(args_string_array, 3, file_type_opt)
```

Comment:
👍🏼 much cleaner. Although I hate on the metals side we have to send it in like this.